upgrades/0.0.105.add-node-volume-encryption/pkg/eksctl/api.go

```diff
@@ -0,0 +1,119 @@
+package eksctl
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"text/template"
+
+	"github.com/oslokommune/okctl-upgrade/upgrades/0.0.105.add-node-volume-encryption/pkg/lib/manifest/apis/okctl.io/v1alpha1"
+)
+
+func GenerateClusterConfig(cluster v1alpha1.Cluster, nodegroupNames []string) (io.Reader, error) {
+	t, err := template.New("clusterconfig").Parse(clusterConfigTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("parsing template: %w", err)
+	}
+
+	buf := bytes.Buffer{}
+
+	err = t.Execute(&buf, clusterConfigTemplateOpts{
+		ClusterName: cluster.Metadata.Name,
+		AccountID:   cluster.Metadata.AccountID,
+		Region:      cluster.Metadata.Region,
+		NodeGroups:  parseNodeGroups(nodegroupNames),
```

Comment:
You don't need to change anything - it's just a comment:

Another approach to generating a cluster config, would be to generate a NodeGroup struct, and from that generate node group names. By doing that the code could possibly be more straight forward I believe, because we could create stuff directly without having to parse strings / node group names in order to build up a struct. (See similar code in https://github.com/oslokommune/okctl/blob/fafebf16566649e6830787782305435178b4515c/pkg/clusterconfig/clusterconfig.go#L143 - though in the upgrade's case it would be less fields).

Reply:
The code in the link expects all regions to have 3 availability zones. I was not sure if I could assume that. Besides, I prefer defining configuration as yaml, not as structs. So I try to define as much as possible as yaml

Reply:
I think I might not have explained myself correctly, as the YAML is not what I'm referring to here. parseNodeGroups currently returns a struct based on parsing a string. I however am suggesting creating the struct that NodeGroups expects directly, instead of doing it as a string in `generateNodegroupNames`.

But probably better to take this over a morning coffee.